src/main/java/frc/robot/scoreStates/ShootState.java

```diff
@@ -2,62 +2,137 @@
-                                                .getRotations(),
-                                        ScoreStateConstants.TURN_TOLERANCE.in(Units.Rotations));
+                        return rotate();
                     }
                     optimalTranslation = botPose.getTranslation().nearest(optimalPoints);
 
                     return AutoBuilder.pathfindToPose(
-                            new Pose2d(
-                                    optimalTranslation,
-                                    Utils.calculateOptimalRotation(
-                                            optimalTranslation, speakerPose)),
+                            new Pose2d(optimalTranslation, new Rotation2d(optimalRotation)),
                             Constants.AUTO_CONSTRAINTS);
                 },
                 Set.of(SwerveDrive.getInstance()));
     }
 
     @Override
-    public Command stateInitialize() {
-        return null;
+    public Command initializeCommand() {
+        botPose = PoseEstimation.getInstance().getEstimatedPose();
+        return Commands.run(
+                () -> {
+                    if (isRed()) {
+                        if (botPose.getX()
+                                < ScoreStateConstants.RED_BOUNDS_MAP.get(botPose.getY()).value) {
+                            inBounds = false;
+                        }
+                        speakerPose = ScoreStateConstants.SPEAKER_POSE_RED;
+                        optimalPoints = ScoreStateConstants.OPTIMAL_POINTS_SHOOT_RED;
+                    } else {
+                        if (botPose.getX()
+                                > ScoreStateConstants.BLUE_BOUNDS_MAP.get(botPose.getY()).value) {
+                            inBounds = false;
+                        }
+                        speakerPose = ScoreStateConstants.SPEAKER_POSE_BLUE;
+                        optimalPoints = ScoreStateConstants.OPTIMAL_POINTS_SHOOT_BLUE;
+                    }
+                    optimalTranslation = botPose.getTranslation().nearest(optimalPoints);
+                    optimalRotation =
+                            Utils.calcRotationToTranslation(optimalTranslation, speakerPose)
+                                    .getRotations();
+                    distanceToSpeaker = Utils.getDistanceFromPoint(speakerPose, botPose);
+                });
+    }
+
+    @Override
+    public Command initializeSubsystems() {
+        shooterToSpeakerHeight =
+                ScoreStateConstants.SPEAKER_TARGET_HEIGHT
+                        - ShooterConstants.SHOOTER_HEIGHT.in(Units.Meters);
+        return Commands.parallel(
+                shooter.setVelocity(
+                        () ->
+                                Units.RotationsPerSecond.of(
+                                                ShooterConstants.VELOCITY_BY_DISTANCE.get(
+                                                                distanceToSpeaker)
```

Comment:
You have to create an `InterpolableDouble` to use the `get` method of the map.

Reply:
1967210079425d3d013e77f10af06ba3fd5a13ed